File: CareConnect/CareConnect.Models/Responses/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace CareConnect.Models.Responses
{
    public class Appointment
    {
        public int AppointmentId { get; set; }

        public int EmployeeAvailabilityId { get; set; }

        public string AppointmentType { get; set; } = null!;

        public int AttendanceStatusId { get; set; }

        public int ClientId { get; set; }

        public int ChildId { get; set; }

        public DateTime Date { get; set; }

        public string? Description { get; set; }

        public string? Note { get; set; }

        public DateTime ModifiedDate { get; set; }

        public string? StateMachine { get; set; }

        public string? PaymentIntentId { get; set; }

        [NotMapped] 
        public bool Paid => PaymentIntentId != null;

        public virtual AttendanceStatus AttendanceStatus { get; set; } = null!;

        public virtual EmployeeAvailability EmployeeAvailability { get; set; } = null!;

        public virtual ClientsChild ClientsChild { get; set; } = null!;

    }
}

File: UI/careconnect_admin/lib/models/responses/appointment.dart
import 'package:careconnect_admin/models/responses/attendance_status.dart';
import 'package:careconnect_admin/models/responses/clients_child.dart';

import 'package:careconnect_admin/models/responses/employee_availability.dart';

import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable(explicitToJson: true)
class Appointment {
  final int appointmentId;
  final int employeeAvailabilityId;
  final String appointmentType;
  final int attendanceStatusId;
  final DateTime date;
  final String? description;
  final String? note;
  String? stateMachine;
  final bool paid;
  final DateTime modifiedDate;
  final AttendanceStatus? attendanceStatus;
  final EmployeeAvailability? employeeAvailability;
  final int clientId;
  final int childId;
  final ClientsChild clientsChild;

  Appointment({
    required this.appointmentId,
    required this.employeeAvailabilityId,
    required this.appointmentType,
    required this.attendanceStatusId,
    required this.date,
    this.description,
    this.note,
    this.stateMachine,
    required this.paid,
    required this.modifiedDate,
    this.attendanceStatus,
    this.employeeAvailability,
    required this.clientId,
    required this.childId,
    required this.clientsChild,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
